drivers/watchdog/wdt_npcx.c

```diff
@@ -0,0 +1,302 @@
+		return -ENOTSUP;
+	}
+
+	if ((options & WDT_OPT_PAUSE_HALTED_BY_DBG) != 0) {
+		LOG_ERR("WDT_OPT_PAUSE_HALTED_BY_DBG is not supported");
+		return -ENOTSUP;
+	}
+
+	/*
+	 * One clock period of T0 timer is 32/32.768 KHz = 0.976 ms.
+	 * Then the counter value is timeout/0.976 - 1.
+	 */
+	inst->TWDT0 = MAX(ceiling_fraction(data->timeout * NPCX_WDT_CLK,
+				32 * 1000) - 1, 1);
+
+	/* Configure 8-bit watchdog counter */
+	inst->WDCNT = MIN(ceiling_fraction(data->timeout, 32) +
+					CONFIG_WDT_NPCX_DELAY_CYCLES, 0xff);
+
+	LOG_DBG("WDT setup: TWDT0, WDCNT are %d, %d", inst->TWDT0, inst->WDCNT);
+
+	key = irq_lock();
+	/* Reload and restart T0 timer */
+	inst->T0CSR |= BIT(NPCX_T0CSR_RST);
+	/* Wait for timer is loaded and restart */
+	while (IS_BIT_SET(inst->T0CSR, NPCX_T0CSR_RST))
+		;
+	irq_unlock(key);
+
+	/* Configure t0 timer interrupt and its isr. */
+	wdt_config_t0out_interrupt(dev);
+
+	/* Enable irq of t0-out expired event */
+	npcx_miwu_irq_enable(&config->t0out);
+
+	return 0;
+}
+
+static int wdt_npcx_disable(const struct device *dev)
+{
+	const struct wdt_npcx_config *const config = DRV_CONFIG(dev);
+	struct wdt_npcx_data *const data = DRV_DATA(dev);
+	struct twd_reg *const inst = HAL_INSTANCE(dev);
+
+	/*
+	 * Ensure we have waited at least 3 watchdog ticks before
+	 * stopping watchdog
+	 */
+	while (k_uptime_get() - data->last_watchdog_touch <
+			NPCX_WDT_MIN_WND_TIME)
+		continue;
+
+	/*
+	 * Stop and unlock watchdog by writing 87h, 61h and 63h
+	 * sequence bytes to WDSDM register
+	 */
+	inst->WDSDM = 0x87;
+	inst->WDSDM = 0x61;
+	inst->WDSDM = 0x63;
+
+	/* Disable irq of t0-out expired event and mark it uninstalled */
+	npcx_miwu_irq_disable(&config->t0out);
+	data->timeout_installed = false;
+
+	return 0;
+}
+
+static int wdt_npcx_feed(const struct device *dev, int channel_id)
+{
+	ARG_UNUSED(channel_id);
+	struct wdt_npcx_data *const data = DRV_DATA(dev);
+	struct twd_reg *const inst = HAL_INSTANCE(dev);
+
+	/* Feed watchdog by writing 5Ch to WDSDM */
+	inst->WDSDM = 0x5C;
+	data->last_watchdog_touch = k_uptime_get();
+
+	return 0;
+}
+
+/* WDT driver registration */
+static const struct wdt_driver_api wdt_npcx_driver_api = {
+	.setup = wdt_npcx_setup,
+	.disable = wdt_npcx_disable,
+	.install_timeout = wdt_npcx_install_timeout,
+	.feed = wdt_npcx_feed,
+};
+
+static int wdt_npcx_init(const struct device *dev)
+{
+	struct twd_reg *const inst = HAL_INSTANCE(dev);
+
+#ifdef CONFIG_WDT_DISABLE_AT_BOOT
+	wdt_npcx_disable(dev);
+#endif
+
+	/*
+	 * TWCFG (Timer Watchdog Configuration) setting
+	 * [7:6]- Reserved = 0
+	 * [5]  - WDSDME   = 1: Feed watchdog by writing 5Ch to WDSDM
+	 * [4]  - WDCT0I   = 1: Select T0IN as watchdog prescaler clock
+	 * [3]  - LWDCNT   = 0: Don't lock WDCNT register
+	 * [2]  - LTWDT0   = 0: Don't lock TWDT0 register
+	 * [1]  - LTWCP    = 0: Don't lock TWCP register
+	 * [0]  - LTWCFG   = 0: Don't lock TWCFG register
+	 */
+	inst->TWCFG = 0x30;
+
```

Comment:
You have already defined nice macros like as NPCX_TWCFG_LTWCFG defined at soc/arm/nuvoton_npcx/common/reg/reg_def.h. Don't you use it?

Reply:
Good idea. I will use them in this line later.

Reply:
Done.